src/server/hook_func.c

```diff
@@ -3150,17 +3148,14 @@ attribute_jobmap_init(job *pjob, struct attribute_jobmap *a_map)
 	}
 
 	for (index = 0; (a_index=(int)a_map[index].attr_i) >= 0; ++index) {
-		if (is_attr_set(&a_map[index].attr_val)) {
-			if (job_attr_def[a_index].at_free) {
-				job_attr_def[a_index].at_free(&a_map[index].attr_val);
-			}
-		}
+		if (is_attr_set(&a_map[index].attr_val))
+			free_attr_generic(job_attr_def, &a_map[index].attr_val, a_index);
```

Comment:
why not just call free_attr_generic -> free_attr ? I don't think we'll ever need free_attr_long, free_attr_char etc., so, i think we can just call it free_attr